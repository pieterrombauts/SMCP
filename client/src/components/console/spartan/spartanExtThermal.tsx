import React, { useEffect, useState, useRef } from 'react';
import styled from 'styled-components';
import { generateRandVal } from '../../../dataGenerator';

interface AppProps {
  className?: string;
};

const defaultValues = {
  Spartan: {
    NH3Flow1: 0,
    NH3Flow2: 0,

    NH3Out1: 0,
    NH3Out2: 0,

    NH3OutTemp1: 0,
    NH3OutTemp2: 0,

    rotDeg1: 0,
    rotDeg2: 0
  }
}

const USpartanExtThermal: React.FC<AppProps> = ( props ) => {

  var [values, setValues] = useState(defaultValues)
  const intervalRef = useRef(0);
  useEffect(() => {
    if (intervalRef.current !== 0) {
      clearInterval(intervalRef.current);
    }
    const id = setInterval(() => {
      var newValues = {
        Spartan: {
          NH3Flow1: generateRandVal(3489, 3500).toFixed(2),
          NH3Flow2: generateRandVal(4400, 4500).toFixed(2),

          NH3Out1: generateRandVal(2100, 2200).toFixed(2),
          NH3Out2: generateRandVal(2200, 2300).toFixed(2),

          NH3OutTemp1: generateRandVal(4, 4.5).toFixed(2),
          NH3OutTemp2: generateRandVal(4.5, 5).toFixed(2),

          rotDeg1: generateRandVal(25, 30).toFixed(2),
          rotDeg2: generateRandVal(-40, -35).toFixed(2),

        }
      }
      setValues(newValues);
    }, 5000);
    intervalRef.current = id;
  },[])

  return(
    <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
      <image x={0} y={0} width={1300} height={990} href={'/media/Spartan_2.png'}></image>
      <rect x="45%" y="22%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#1C4B5D"}} />
      <text x="46%" y="24%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.NH3Flow1}</text>
      <rect x="55%" y="22%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#1C4B5D"}} />
      <text x="56%" y="24%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.NH3Flow2}</text>

      <rect x="45%" y="25.75%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#1C4B5D"}} />
      <text x="46%" y="27.75%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.NH3Out1}</text>
      <rect x="55%" y="25.75%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#1C4B5D"}} />
      <text x="56%" y="27.75%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.NH3Out2}</text>

      <rect x="45%" y="29.5%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#1C4B5D"}} />
      <text x="46%" y="31.5%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.NH3OutTemp1}</text>
      <rect x="55%" y="29.5%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#1C4B5D"}} />
      <text x="56%" y="31.5%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.NH3OutTemp2}</text>
      
      <rect x="45%" y="33.25%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#1C4B5D"}} />
      <text x="46%" y="35.25%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.rotDeg1}</text>
      <rect x="55%" y="33.25%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#1C4B5D"}} />
      <text x="56%" y="35.25%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.rotDeg2}</text>  
    </svg>
  );
}

export const SpartanExtThermal = styled(USpartanExtThermal)`
  width: 100%;
  height: 100%;
  position: absolute;

  #spartan-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  #spartan-buttons .nav {
    margin-left: 20px;
    max-width: 300px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }

  #spartan-buttons .nav-item a {
    padding: 20px 30px;
  }
`;