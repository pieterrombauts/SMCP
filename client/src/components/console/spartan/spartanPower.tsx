import React, { useEffect, useState, useRef } from 'react';
import styled from 'styled-components';
import { generateRandVal } from '../../../dataGenerator';


interface SpartanPowerInt {
  className?: string;
};

const defaultValues = {
  Spartan: {
    voltage2b: 0,
    current2b: 0,
    array2b: 0,

    voltage4b: 0,
    current4b: 0,
    array4b: 0,

    voltage4a: 0,
    current4a: 0,
    array4a: 0,

    voltage2a: 0,
    current2a: 0,
    array2a: 0,

    voltage1a: 0,
    current1a: 0,
    array1a: 0,

    voltage3a: 0,
    current3a: 0,
    array3a: 0,

    voltage3b: 0,
    current3b: 0,
    array3b:0,

    voltage1b:  0,
    current1b:  0,
    array1b:  0
  }
}

const USpartanPower: React.FC<SpartanPowerInt> = ( props ) => {
  var [values, setValues] = useState(defaultValues)
  const intervalRef = useRef(0);
  useEffect(() => {
    if (intervalRef.current !== 0) {
      clearInterval(intervalRef.current);
    }
    const id = setInterval(() => {
      var newValues = {
        Spartan: {
          voltage2b: generateRandVal(158, 160).toFixed(2),
          current2b: generateRandVal(-45, -43).toFixed(2),
          array2b: generateRandVal(335, 340).toFixed(2),

          voltage4b: generateRandVal(161, 163).toFixed(2),
          current4b: generateRandVal(-37, -35).toFixed(2),
          array4b: generateRandVal(155, 160).toFixed(2),

          voltage4a: generateRandVal(160, 162).toFixed(2),
          current4a: generateRandVal(-37, -35).toFixed(2),
          array4a: generateRandVal(340, 345).toFixed(2),

          voltage2a: generateRandVal(150, 152).toFixed(2),
          current2a: generateRandVal(-26, -23).toFixed(2),
          array2a: generateRandVal(20, 25).toFixed(2),

          voltage1a: generateRandVal(161, 163).toFixed(2),
          current1a: generateRandVal(-29, -27).toFixed(2),
          array1a: generateRandVal(160, 165).toFixed(2),

          voltage3a:  generateRandVal(161, 163).toFixed(2),
          current3a: generateRandVal(-29, -27).toFixed(2),
          array3a: generateRandVal(160, 165).toFixed(2),

          voltage3b:  generateRandVal(161, 163).toFixed(2),
          current3b: generateRandVal(-29, -27).toFixed(2),
          array3b: generateRandVal(155, 160).toFixed(2),

          voltage1b:  generateRandVal(161, 163).toFixed(2),
          current1b:  generateRandVal(-51, -49).toFixed(2),
          array1b:  generateRandVal(210, 230).toFixed(2)
        }
      }
      setValues(newValues);
    }, 5000);
    intervalRef.current = id;
  },[])

  return(
    <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
      <image x={0} y={0} width={1300} height={990} href={'/media/Spartan_1.png'}></image>
      <rect x="21%" y="32%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="20.5%" y="34.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.voltage2b}</text>
      <rect x="21%" y="37%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="20.5%" y="39.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.current2b}</text>
      <rect x="21%" y="42%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="20.5%" y="44.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.array2b}</text>

      <rect x="21%" y="72%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="20.5%" y="74.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.voltage4b}</text>
      <rect x="21%" y="77%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="20.5%" y="79.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.current4b}</text>
      <rect x="21%" y="82%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="20.5%" y="84.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.array4b}</text> 


      <rect x="36%" y="32%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="35.8%" y="34.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.voltage4b}</text>
      <rect x="36%" y="37%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="35.8%" y="39.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.current4b}</text>
      <rect x="36%" y="42%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="35.8%" y="44.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.array4b}</text>
      
      <rect x="36%" y="72%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="35.8%" y="74.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.voltage2a}</text>
      <rect x="36%" y="77%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="35.8%" y="79.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.current2a}</text>
      <rect x="36%" y="82%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="35.8%" y="84.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.array2a}</text>
      

      <rect x="76.5%" y="32%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="76.5%" y="34.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.voltage1a}</text>
      <rect x="76.5%" y="37%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="76.5%" y="39.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.current1a}</text>
      <rect x="76.5%" y="42%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="76.5%" y="44.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.array1a}</text> 
      
      <rect x="76.5%" y="72%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="76.5%" y="74.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.voltage3a}</text>
      <rect x="76.5%" y="77%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="76.5%" y="79.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.current3a}</text>
      <rect x="76.5%" y="82%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="76.5%" y="84.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.array3a}</text>


      <rect x="92%" y="32%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="92%" y="34.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.voltage3b}</text>
      <rect x="92%" y="37%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="92%" y="39.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.current3b}</text>
      <rect x="92%" y="42%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="92%" y="44.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.array3b}</text> 
      
      <rect x="92%" y="72%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="92%" y="74.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.voltage1b}</text>
      <rect x="92%" y="77%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="92%" y="79.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.current1b}</text>
      <rect x="92%" y="82%" rx="5" ry="5" width="8%" height="3%" style={{fill:"#000000"}} />
      <text x="92%" y="84.4%" fill="#d6ab62" fontSize="17" fontFamily="Verdana">{values.Spartan.array1b}</text>   
    </svg>
  );
}

export const SpartanPower = styled(USpartanPower)`
  width: 100%;
  height: 100%;
  position: absolute;

  #spartan-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  #spartan-buttons .nav {
    margin-left: 20px;
    max-width: 300px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }

  #spartan-buttons .nav-item a {
    padding: 20px 30px;
  }
`;
