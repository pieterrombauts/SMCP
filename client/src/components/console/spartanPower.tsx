import React, { useEffect, useState, useRef } from 'react';
import styled from 'styled-components';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Tab from 'react-bootstrap/Tab';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import spartanPowerCrew from "./../../media/spartanPowerCrew.png";
import { generateRandVal } from '../../dataGenerator';


interface SpartanPowerInt {
  className?: string;
};

const defaultValues = {
  Spartan: {
    voltage1: 0,
    voltage2: 0,
    voltage3: 0,
    voltage4: 0,
    current1: 0,
    current2: 0,
    current3: 0,
    current4: 0,
    array1: 0,
    array2: 0,
    array3: 0,
    array4: 0,

    voltage1b: 0,
    voltage2b: 0,
    voltage3b: 0,
    voltage4b: 0,
    current1b: 0,
    current2b: 0,
    current3b: 0,
    current4b: 0,
    array1b: 0,
    array2b: 0,
    array3b: 0,
    array4b: 0
    
  }
}

const USpartanPower: React.FC<SpartanPowerInt> = ( props ) => {
  var [values, setValues] = useState(defaultValues)
  const intervalRef = useRef(0);
  useEffect(() => {
    if (intervalRef.current != 0) {
      clearInterval(intervalRef.current);
    }
    const id = setInterval(() => {
      var newValues = {
        Spartan: {
          voltage1: generateRandVal(158, 160).toFixed(2),
          voltage2: generateRandVal(160, 162).toFixed(2),
          voltage3: generateRandVal(161, 163).toFixed(2),
          voltage4: generateRandVal(161, 163).toFixed(2),

          current1: generateRandVal(-45, -43).toFixed(2),
          current2: generateRandVal(-37, -35).toFixed(2),
          current3: generateRandVal(-29, -27).toFixed(2),
          current4: generateRandVal(-39, -37).toFixed(2),

          array1: generateRandVal(335, 340).toFixed(2),
          array2: generateRandVal(340, 345).toFixed(2),
          array3: generateRandVal(160, 165).toFixed(2),
          array4: generateRandVal(155, 160).toFixed(2),

          voltage1b: generateRandVal(150, 152).toFixed(2),
          voltage2b: generateRandVal(162, 162).toFixed(2),
          voltage3b: generateRandVal(161, 163).toFixed(2),
          voltage4b: generateRandVal(161, 163).toFixed(2),

          current1b: generateRandVal(-40, -38).toFixed(2),
          current2b: generateRandVal(-26, -23).toFixed(2),
          current3b: generateRandVal(-46, -44).toFixed(2),
          current4b: generateRandVal(-51, -49).toFixed(2),

          array1b: generateRandVal(20, 25).toFixed(2),
          array2b: generateRandVal(20, 25).toFixed(2),
          array3b: generateRandVal(200, 220).toFixed(2),
          array4b: generateRandVal(210, 230).toFixed(2)
        }
      }
      setValues(newValues);
    }, 5000);
    intervalRef.current = id;
  },[])

  return(
  
      <Container style={{paddingTop:"20%"}}>
        <Row>
          <Col style={{border:"white solid 1px", color:"white"}}>Channel Name</Col>
          <Col style={{border:"white solid 1px", color:"white"}}>2B</Col>
          <Col xs={5} style={{border:"white solid 1px", color:"white"}}>4A</Col>
          <Col style={{border:"white solid 1px", color:"white"}}>1A</Col>
          <Col style={{border:"white solid 1px", color:"white"}}>3B</Col>
        </Row>
        <Row>
          <Col style={{border:"white solid 1px", color:"white"}}>Voltage (V)</Col>
          <Col id='voltage1' style={{border:"white solid 1px", color:"gold"}}>  {values.Spartan.voltage1}</Col>
          <Col xs={5} style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.voltage2}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.voltage3}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.voltage4}</Col>
        </Row>

        <Row>
          <Col style={{border:"white solid 1px", color:"white"}}>Current (A)</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.current1}</Col>
          <Col xs={5} style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.current2}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.current3}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.current4}</Col>
        </Row>

        <Row>
          <Col style={{border:"white solid 1px", color:"white"}}>Array Position (degrees)</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.array1} </Col>
          <Col xs={5} style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.array2}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.array3}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.array4}</Col>
        </Row>

        <Row>
          <img src={spartanPowerCrew} style={{width:"950px", height:"50px"}}/> 
        </Row>

        
      
        <Row>
          <Col style={{border:"white solid 1px", color:"white"}}>Channel Name</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> 4B</Col>
          <Col xs={5} style={{border:"white solid 1px", color:"gold"}}>2A</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> 3A</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> 1B</Col>
        </Row>

        <Row>
          <Col style={{border:"white solid 1px", color:"white"}}>Voltage (V)</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.voltage1b}</Col>
          <Col xs={5} style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.voltage2b}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.voltage3b}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.voltage4b}</Col>
        </Row>

        <Row>
          <Col style={{border:"white solid 1px", color:"white"}}>Current (A)</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.current1b}</Col>
          <Col xs={5} style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.current2b}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.current3b}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.current4b}</Col>
        </Row>

        <Row>
          <Col style={{border:"white solid 1px", color:"white"}}>Array Position (degrees)</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.array1b}</Col>
          <Col xs={5} style={{border:"white solid 1px", color:"gold"}}> {values.Spartan.array2b}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}>{values.Spartan.array3b}</Col>
          <Col style={{border:"white solid 1px", color:"gold"}}>{values.Spartan.array4b}</Col>
        </Row>

       </Container>
         
  );
}

export const SpartanPower = styled(USpartanPower)`
  width: 100%;
  height: 100%;
  position: absolute;

  #spartan-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  #spartan-buttons .nav {
    margin-left: 20px;
    max-width: 300px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }

  #spartan-buttons .nav-item a {
    padding: 20px 30px;
  }
`;
