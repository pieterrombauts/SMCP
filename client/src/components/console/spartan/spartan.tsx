import React, { useState } from 'react';
import styled from 'styled-components';
import Nav from 'react-bootstrap/Nav';
import Tab from 'react-bootstrap/Tab';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import {SpartanExtThermal} from "./spartanExtThermal";
import {SpartanPower} from "./spartanPower";
import socket from '../../Socket';

interface AppProps {
  className?: string;
};

const USpartan: React.FC<AppProps> = ( props ) => {
  const [ firstOpensETC, setFirstOpensETC ] = useState<boolean>(true);
  function updateFirstConsoleETCOpen(event: React.MouseEvent<any>) {
    if (firstOpensETC) {
      socket.emit('FIRST_CONSOLE_OPEN', 'spartan-etc')
      setFirstOpensETC(false);
    }
  }
  return(
    <Tab.Container id="spartan-tabs" defaultActiveKey="power">
      <Row className={props.className}>
        <Col id="spartan-content" sm={10}>
          <Tab.Content>
            <Tab.Pane eventKey="power">
              < SpartanPower />
            </Tab.Pane>
            <Tab.Pane eventKey="ext-thermal">
              < SpartanExtThermal />
            </Tab.Pane>
          </Tab.Content>
        </Col>
        <Col id="spartan-buttons" sm={2}>
          <Nav variant="pills" className="flex-column">
            <Nav.Item>
              <Nav.Link eventKey="power">POWER CONSOLE</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link onClick={updateFirstConsoleETCOpen} eventKey="ext-thermal">EXT. THERMAL CONTROL</Nav.Link>
            </Nav.Item>
          </Nav>
        </Col>
      </Row>
    </Tab.Container>
  );
}

export const Spartan = styled(USpartan)`
  width: 100%;
  height: 100%;
  position: absolute;

  #spartan-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  #spartan-buttons .nav {
    margin-left: 20px;
    max-width: 300px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }

  #spartan-buttons .nav-item a {
    padding: 20px 30px;
  }
`;