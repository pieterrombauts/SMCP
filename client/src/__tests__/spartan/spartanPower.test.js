import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { SpartanPower } from '../../components/console/spartan/spartanPower'
import { shallow, mount, render} from 'enzyme';

describe("Spartan Power display", () => {
  it('renders correctly', () => {
    const wrapper = mount(<SpartanPower />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})