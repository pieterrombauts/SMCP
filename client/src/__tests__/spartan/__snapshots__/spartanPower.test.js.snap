// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartan Power display renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  position: absolute;
}

.c0 #spartan-buttons {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 #spartan-buttons .nav {
  margin-left: 20px;
  max-width: 300px;
  background-color: #f8f9fa;
  border-radius: 3px;
}

.c0 #spartan-buttons .nav-item a {
  padding: 20px 30px;
}

<Styled(USpartanPower)>
  <USpartanPower
    className="c0"
  >
    <svg
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 1300 990"
    >
      <image
        height={990}
        href="/media/Spartan_1.png"
        width={1300}
        x={0}
        y={0}
      />
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="21%"
        y="32%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="20.5%"
        y="34.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="21%"
        y="37%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="20.5%"
        y="39.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="21%"
        y="42%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="20.5%"
        y="44.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="21%"
        y="72%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="20.5%"
        y="74.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="21%"
        y="77%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="20.5%"
        y="79.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="21%"
        y="82%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="20.5%"
        y="84.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="36%"
        y="32%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="35.8%"
        y="34.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="36%"
        y="37%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="35.8%"
        y="39.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="36%"
        y="42%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="35.8%"
        y="44.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="36%"
        y="72%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="35.8%"
        y="74.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="36%"
        y="77%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="35.8%"
        y="79.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="36%"
        y="82%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="35.8%"
        y="84.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="76.5%"
        y="32%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="76.5%"
        y="34.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="76.5%"
        y="37%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="76.5%"
        y="39.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="76.5%"
        y="42%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="76.5%"
        y="44.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="76.5%"
        y="72%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="76.5%"
        y="74.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="76.5%"
        y="77%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="76.5%"
        y="79.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="76.5%"
        y="82%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="76.5%"
        y="84.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="92%"
        y="32%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="92%"
        y="34.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="92%"
        y="37%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="92%"
        y="39.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="92%"
        y="42%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="92%"
        y="44.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="92%"
        y="72%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="92%"
        y="74.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="92%"
        y="77%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="92%"
        y="79.4%"
      >
        0
      </text>
      <rect
        height="3%"
        rx="5"
        ry="5"
        style={
          Object {
            "fill": "#000000",
          }
        }
        width="8%"
        x="92%"
        y="82%"
      />
      <text
        fill="#d6ab62"
        fontFamily="Verdana"
        fontSize="17"
        x="92%"
        y="84.4%"
      >
        0
      </text>
    </svg>
  </USpartanPower>
</Styled(USpartanPower)>
`;
